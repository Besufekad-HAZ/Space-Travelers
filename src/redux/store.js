// import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import {
  legacy_createStore as createStore,
  combineReducers,
  applyMiddleware,
} from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketsSlice';
import missionsSlice from './missions/slice-Missions';

const reducers = combineReducers({
  rockets: rocketsSlice,
  missions: missionsSlice,
});

const store = createStore(reducers, applyMiddleware(thunk));
export default store;
