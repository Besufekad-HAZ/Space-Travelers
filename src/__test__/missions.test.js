import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Mission from '../component/missions';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Mission component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Mission />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
