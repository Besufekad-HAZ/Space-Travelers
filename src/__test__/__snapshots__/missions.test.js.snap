// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component renders correctly 1`] = `
<tr>
  <td />
  <td />
  <td
    className="button-status"
  >
    <div
      className="not_active"
    >
      NOT A MEMBER
    </div>
  </td>
  <td
    className="button-join"
  >
    <button
      className="join"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
